src/cli/options/options.go

```diff
@@ -50,10 +51,40 @@ var (
 	skipReduceFV         bool
 )
 
+// s3 bucket info from flags
+var (
+	Bucket          string
+	Endpoint        string
+	Prefix          string
+	DoNotUseTLS     bool
+	DoNotVerifyTLS  bool
+	AccessKey       string
+	SecretAccessKey string
+	SessionToken    string
+)
+
 // AddGlobalOperationFlags adds the global operations flag set.
 func AddGlobalOperationFlags(cmd *cobra.Command) {
 	fs := cmd.PersistentFlags()
 	fs.BoolVar(&noStatsFV, NoStatsFN, false, "disable anonymous usage statistics gathering")
+
+	// Flags addition ordering should follow the order we want them to appear in help and docs:
+	// More generic and more frequently used flags take precedence.
+
+	// S3 flags
```

Comment:
s3 and m365 are not global implementations.  In the future we will inevitably support other systems, and the declaration of these flags within a global set will cause more confusion/errors than solution.  Another reason to move these declarations to a different owner.